src/zenml/config/step_configurations.py

```diff
@@ -26,14 +26,12 @@
 class PartialArtifactConfiguration(StrictBaseModel):
     """Class representing a partial input/output artifact configuration."""
 
-    artifact_source: Optional[str] = None
```

Comment:
This is not backwards compatible and will crash when loading an old config from the zen store in e.g. post execution

Reply:
I guess we could explicitly pop `artifact_source` before we create the `Step` in `StepRunSchema.to_model()`. What do you think about that?

Reply:
Added a pydantic root validator to remove the `artifact_source`